src/main/haskell/kore/src/Kore/IndexedModule/MetadataTools.hs

```diff
@@ -10,52 +10,73 @@ Portability : portable
 -}
 module Kore.IndexedModule.MetadataTools
     ( MetadataTools (..)
-    , SortTools
+    , SymbolOrAliasSorts
     , extractMetadataTools
     , getResultSort
     ) where
 
+import           Data.Graph
+import qualified Data.Map as Map
+
 import Kore.AST.Common
 import Kore.AST.MetaOrObject
 import Kore.ASTHelpers
+       ( ApplicationSorts (..) )
+import Kore.Attribute.Subsort
 import Kore.IndexedModule.IndexedModule
 import Kore.IndexedModule.Resolvers
 
 -- |'MetadataTools' defines a dictionary of functions which can be used to
 -- access the metadata needed during the unification process.
 data MetadataTools level attributes = MetadataTools
     { symAttributes :: SymbolOrAlias level -> attributes
+    -- ^ get the attributes of a symbol or alias
     , sortAttributes :: Sort level -> attributes
-    , sortTools  :: SortTools level
+    -- ^ get the attributes of a sort
+    , symbolOrAliasSorts :: SymbolOrAliasSorts level
+    -- ^ get the signature of a symbol or alias
     , isSubsortOf :: Sort level -> Sort level -> Bool
+    {- ^ @isSubsortOf a b@ is true if sort @a@ is a subsort of sort @b@,
+       including when @a@ equals @b@. -}
     }
 
--- TODO: Rename this as `SortGetter` or something similar, `Tools` is
--- too general.
-type SortTools level = SymbolOrAlias level -> ApplicationSorts level
+type SymbolOrAliasSorts level = SymbolOrAlias level -> ApplicationSorts level
 
 -- |'extractMetadataTools' extracts a set of 'MetadataTools' from a
 -- 'KoreIndexedModule'.  The metadata tools are functions yielding information
 -- about application heads, such as its attributes or
 -- its argument and result sorts.
 --
 extractMetadataTools
-    :: MetaOrObject level
+    ::forall level atts . MetaOrObject level
     => KoreIndexedModule atts
     -> MetadataTools level atts
 extractMetadataTools m =
   MetadataTools
     { symAttributes = getHeadAttributes m
     , sortAttributes = getSortAttributes m
-    , sortTools  = getHeadApplicationSorts m
-    -- TODO: Implement.
-    , isSubsortOf = const $ const $ False
+    , symbolOrAliasSorts  = getHeadApplicationSorts m
+    , isSubsortOf = checkSubsort
     }
+  where
+    checkSubsort = case isMetaOrObject @level [] of
+        IsMeta -> (==)
+        IsObject -> let
+              subsortTable = Map.unionsWith (++)
+                  [ Map.insert subsort [] $ Map.singleton supersort [subsort]
+                  | Subsort subsort supersort <- indexedModuleSubsorts m]
+              (sortGraph, _, getSortId) =
+                  graphFromEdges [((),supersort,subsorts)
+                                 |(supersort,subsorts)
```

Comment:
Here you don't have a space after |, above you have one, consistency would be nice (I prefer having a space).